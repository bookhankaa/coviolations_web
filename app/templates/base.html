<!DOCTYPE html>
{% load static %}
<html>
<head>
    <meta charset="utf-8" />
    <title>coviolations/{% block title %}{% endblock %}</title>
    <link
        rel="stylesheet"
        href="{% static 'bootstrap/dist/css/bootstrap.min.css' %}"
    />
    <link
        rel="stylesheet"
        href="{% static 'font-awesome/css/font-awesome.min.css' %}"
    />
    <script
        type="text/javascript"
        src='{% static 'jquery/jquery.min.js' %}'
    ></script>
    <script
        type="text/javascript"
        src='{% static 'bootstrap/dist/js/bootstrap.min.js' %}'
    ></script>
</head>
<body>
    <h1>Coviolations</h1>
    {% block content %}
    {% endblock %}
</body>
</html>