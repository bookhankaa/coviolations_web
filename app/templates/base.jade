doctype 5
html(lang='ru')
    head
        title coviolations.io
        link(rel='stylesheet', href='/static/bootstrap/dist/css/bootstrap.css')
        link(rel='stylesheet', href='/static/font-awesome/css/font-awesome.min.css')
        link(rel='stylesheet', href='/static/ngprogress/ngProgress.css')
        link(rel='stylesheet', href='/static/styles/main.css')
        script(type='text/javascript').
            var successTasks = {{ success }};
            var failedTasks = {{ failed }};
            var isAuthenticated = {% if user.is_authenticated %} true {% else %} false {% endif %};
            var domain = '{{ SITE.domain }}';
    body
        .container(ng-view)
    script(data-main='/static/scripts/main', src='/static/requirejs/require.js')
