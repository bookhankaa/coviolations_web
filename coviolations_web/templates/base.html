<!DOCTYPE html><html lang="ru"><head><title>coviolations.io</title><link rel="icon" type="image/png" href="/static/images/favicon.png"><link rel="stylesheet" href="/static/bootstrap/dist/css/bootstrap.css"><link rel="stylesheet" href="/static/font-awesome/css/font-awesome.min.css"><link rel="stylesheet" href="/static/ngprogress/ngProgress.css"><link rel="stylesheet" href="/static/google-code-prettify/distrib/google-code-prettify/prettify.css"><link rel="stylesheet" href="/static/nvd3/nv.d3.min.css"><link rel="stylesheet" href="/static/styles/main.css"><script type="text/javascript">var successTasks = {{ success }};
var failedTasks = {{ failed }};
var isAuthenticated = {% if user.is_authenticated %} true {% else %} false {% endif %};
var domain = '{{ site.domain }}';
var debug = {% if debug %} true {% else %} false {% endif %};
{% if user.is_authenticated %}
    var user = '{{ user.username }}';
{% endif %}</script></head><body><div ng-view class="container"></div></body><script data-main="/static/scripts/main" src="/static/requirejs/require.js"></script></html>