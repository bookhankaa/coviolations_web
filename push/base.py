import json
import redis
from django.conf import settings


class PushSender(object):
    """Push sender"""

    @property
    def r(self):
        """Init redis connection"""
        if not hasattr(self, '_r'):
            self._r = redis.Redis()
        return self._r

    def send(self, **msg):
        """Send message to redis channel"""
        self.r.publish(settings.REDIS_PUSH, json.dumps(msg))


sender = PushSender()
