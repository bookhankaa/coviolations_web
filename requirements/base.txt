django
django-bower
south
django-extensions
django-discover-runner
pillow
django-rq
redis
PyGithub
django-social-auth
oauth2
django-braces
django-tastypie
mimeparse
