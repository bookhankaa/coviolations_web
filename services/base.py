from tools.library import BaseLibrary
from .exceptions import ServiceDoesNotExists


class ServicesLibrary(BaseLibrary):
    """Services library"""
    exception = ServiceDoesNotExists

library = ServicesLibrary()
