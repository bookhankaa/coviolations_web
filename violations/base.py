from functools import partial
from django.conf import settings
from .exceptions import ViolationDoesNotExists


class ViolationsLibrary(object):
    """Violations library"""

    def __init__(self):
        self._violations = {}

    def _register_fnc(self, name, fnc):
        if name in settings.VIOLATIONS:
            self._violations[name] = fnc
        return fnc

    def register(self, name):
        """Register violation in library"""
        return partial(self._register_fnc, name)

    def get(self, name):
        """Get violation"""
        try:
            return self._violations[name]
        except KeyError:
            raise ViolationDoesNotExists(name)


library = ViolationsLibrary()
