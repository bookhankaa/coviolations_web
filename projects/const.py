TREND_CHUNK = 100
