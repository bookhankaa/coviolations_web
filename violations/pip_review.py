from django.template.loader import render_to_string
from tasks.const import STATUS_SUCCESS, STATUS_FAILED
from .base import library


@library.register('pip_review')
def pip_review_violation(data):
    """pip-review violation parser

    :param data: task data
    :type data: dict
    :returns: dict
    """
    count = len(data['raw'].split('\n')) - 1
    data['status'] = STATUS_SUCCESS if count == 0 else STATUS_FAILED
    data['preview'] = render_to_string('violations/pip_review/preview.html', {
        'count': count,
    })
    data['prepared'] = render_to_string(
        'violations/pip_review/prepared.html', {'raw': data['raw']},
    )
    data['plot'] = {
        'outdated': count,
    }
    return data
