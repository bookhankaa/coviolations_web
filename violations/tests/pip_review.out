Django==1.5.3 is available (you have 1.5.2)
amqp==1.2.1 is available (you have 1.0.13)
celery==3.0.23 is available (you have 3.0.22)
django-celery==3.0.23 is available (you have 3.0.21)
django-extensions==1.2.0 is available (you have 1.1.1)
django-userena==1.2.3 is available (you have 1.2.1)
ipython==1.1.0 is available (you have 1.0.0)
kombu==2.5.14 is available (you have 2.5.13)
mimeparse==0.1.4 is available (you have 0.1.3)
pytz==2013d is available (you have 2013b)
redis==2.8.0 is available (you have 2.7.6)
six==1.4.1 is available (you have 1.3.0)
